contracts/oracle/src/contract/feed.rs

```diff
@@ -77,20 +83,88 @@ where
         Ok(prices)
     }
 
-    fn calc_all_prices(
-        &self,
-        storage: &dyn Storage,
+    pub fn all_prices_iter<'a>(
```

Comment:
think about unifying the implementation of this use-case and the previous one for specific currencies done with `fn calc_prices()`

Reply:
@go2bodhi please add a ticket for that and link it here

Reply:
#38